build.sh

```diff
@@ -6,22 +6,23 @@ set -ev
 export PATH=$PATH:~/.cargo/bin:$GOROOT/bin
 export PYENV_VERSION=3.8
 
-run() {
+run_from_diff() {
     echo "$1" | grep "day-" | cut -d "/" -f1 | cut -d "-" -f2 | sort | uniq | xargs -I{} ./aoc run -fntd {}
 }
 
+run_from_inputs() {
+    for day in $1; do echo "$day"; done | sort | uniq | xargs -I{} ./aoc run -fntd {}
+}
 
-# In PRs, GITHUB_HEAD_REF is set the the name of the branch
-# This variable isn't set on regular commits and as we only allowed the CI to run on pushs only in master
-# checking if the variable is the empty string is the same as checking if the branch is master
 # Note: we cannot use "git branch --show-current" as GitHub rewrites the history in actions
-if [ "$GITHUB_HEAD_REF" == "" ];
+if [ "$GITHUB_EVENT_NAME" == "workflow_dispatch" ];
+then
+    run_from_inputs "$INPUTS_DAYS"
+elif [ "$GITHUB_EVENT_NAME" == "pull_request" ];
```

Comment:
This is meant to replace `if [ "$GITHUB_HEAD_REF" == "" ];`
It seems to be equivalent, although not 100% sure